### PR TITLE
feat: add non-root user to docker image

### PR DESCRIPTION
**Pull request by nrwiersma**
## What is the problem I am trying to address?

In many modern deployment environments, root containers are not allowed. While it is simple to switch the user on the fly, that user needs to have a home directory for this to work.

## How is the fix applied?

A non-root user `athens` uid/gid 1000 is created in the docker image, but not used by default. As existing users will still want to use `root` for this to be non-breaking, the new user is not used by default. Documentation on the non-root user has also been added to make this change discoverable.

### CODE REVIEW

File: cmd/proxy/Dockerfile [hunk 1]
```diff
@@ -32,6 +32,9 @@ RUN chmod 644 /config/config.toml
 RUN apk add --update git git-lfs mercurial openssh-client subversion procps fossil tini && \
 	mkdir -p /usr/local/go
 
+ARG USER=athens
+RUN adduser -D -h /home/$USER $USER
```
Comment (manugupt1) [inline cmd/proxy/Dockerfile:36]: It might break k8s and docker-compose which provide external options to control non root access.

Comment (nrwiersma) [reply cmd/proxy/Dockerfile:36]: k8s just lets you set the user id and group id to be run, much like docker does. Should you however need to use `.netrc` or `.gitconfig` you cannot without the user you are using having a home directory.

**Review by DrPsychick — CHANGES_REQUESTED** [review]
Typo "authens"

File: cmd/proxy/Dockerfile (hunk 1 above)
Comment (manugupt1) [reply cmd/proxy/Dockerfile:36]: Makes sense! Should we create non root group as well

Comment (nrwiersma) [reply cmd/proxy/Dockerfile:36]: This is already done by `adduser`. The group is 1000 called `athens`.

**Review by manugupt1 — APPROVED** [review]

**Review by DrPsychick — APPROVED** [review]

**Pull request merged**